slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/validator/BlockValidator.java

```diff
@@ -32,18 +34,24 @@
   private final UInt64 blockSlot;
   private final int validatorId;
   private final SignedBlocksDao signedBlocksDao;
+  private final LowWatermarkDao lowWatermarkDao;
+
+  private final Optional<SigningWatermark> watermark;
 
   public BlockValidator(
       final Handle handle,
       final Bytes signingRoot,
       final UInt64 blockSlot,
       final int validatorId,
-      final SignedBlocksDao signedBlocksDao) {
+      final SignedBlocksDao signedBlocksDao,
+      final LowWatermarkDao lowWatermarkDao) {
     this.handle = handle;
     this.signingRoot = signingRoot;
     this.blockSlot = blockSlot;
     this.validatorId = validatorId;
     this.signedBlocksDao = signedBlocksDao;
+    this.lowWatermarkDao = lowWatermarkDao;
+    watermark = lowWatermarkDao.findLowWatermarkForValidator(handle, validatorId);
```

Comment:
Don't like this happening in the constructor. perhaps use a memoize so it can be done when needed or just pass it in?

Reply:
changed to memoizing